libr/core/disasm.c

```diff
@@ -1283,7 +1283,10 @@ static void ds_build_op_str(RDisasmState *ds, bool print_color) {
 		char **wc_array = r_str_argv (wcdata, &argc);
 		for (i = 0; i < argc; i++) {
 			bgcolor = strchr (wc_array[i], '\x1b');
-			word = r_str_newlen (wc_array[i], bgcolor - wc_array[i]);
+			word = NULL;
+			if (bgcolor - wc_array[i]) {
```

Comment:
what about using `int len = bgcolor - wc_array[i]` and then compare and use later instead of computing the value twice? same for the other cases